### PR TITLE
Remove extensions manager midas3 support & update qSlicerExtensionsManagerModelTest

### PR DESCRIPTION
**Pull request by jcfr**
* [x] Add `Girder_v1` json files to https://github.com/Slicer/Slicer/tree/master/Base/QTCore/Testing/Data/Input
* [x] Upload corresponding extension  to https://github.com/Slicer/SlicerTestingData
* [x] Update [Base/QTCore/Testing/Cxx/CMakeLists.txt](https://github.com/Slicer/Slicer/blob/e2ea8afa61a4d4b0372f3650c7843c56b757a20c/Base/QTCore/Testing/Cxx/CMakeLists.txt#L9-L41) to download extension test packages
* [x] Update `qSlicerExtensionsManagerModelTest` to run test based on `serverAPI`

### CODE REVIEW

**Comment by jcfr** [comment]
After finalizing the integration of this pull request, update needed to resolve https://github.com/Slicer/Slicer/issues/5470#issuecomment-993793610 could be worked on.

**Comment by lassoan** [comment]
@jcfr is this in "draft" state because you plan to push more commits or it is ready for review?

**Comment by jcfr** [comment]
Because it needs to be further improved. We need to first fix the extension manager model test (see referenced PR).

If someone has bandwidth to improve fix the test before I do, it would be great too.

File: Base/QTCore/qSlicerExtensionsManagerModel.h [hunk 1]
```diff
@@ -142,7 +141,7 @@ class Q_SLICER_BASE_QTCORE_EXPORT qSlicerExtensionsManagerModel : public QObject
   /// \sa installExtension, isExtensionInstalled, installedExtensions, extensionInstalled
   int numberOfInstalledExtensions()const;
 
-  /// \brief Return names of all installed extensions
+  /// \brief Return names of all installed extensions sorted in alphapeicall order.
```
Comment (lassoan) [inline Base/QTCore/qSlicerExtensionsManagerModel.h:144]:
```suggestion
  /// \brief Return names of all installed extensions sorted in alphabetical order.
```

File: Base/QTCore/qSlicerExtensionsManagerModel.h [hunk 2]
```diff
@@ -179,7 +178,7 @@ class Q_SLICER_BASE_QTCORE_EXPORT qSlicerExtensionsManagerModel : public QObject
   /// \sa uninstallScheduledExtensions();
   Q_INVOKABLE bool isExtensionScheduledForUninstall(const QString& extensionName)const;
 
-  /// \brief Return names of all enabled extensions
+  /// \brief Return names of all enabled extensions sorted in alphapeicall order.
```
[the remark this answers is not in the document]
Comment (jamesobutler) [reply Base/QTCore/qSlicerExtensionsManagerModel.h:181]: I'm surprised `codespell` didn't catch this.
```suggestion
  /// \brief Return names of all enabled extensions sorted in alphabetical order.
```

File: Base/QTCore/qSlicerExtensionsManagerModel.h (hunk 1 above)
Comment (jcfr) [reply Base/QTCore/qSlicerExtensionsManagerModel.h:144]: Re: replace the extensions manager model with the content of #6417

Exactly I started on this.

**Pull request merged**